src/connector/src/test_data/union-schema.avsc

```diff
@@ -0,0 +1,87 @@
+{
+  "name": "test_student",
+  "type": "record",
+  "fields": [
+    {
+      "name": "id",
+      "type": "int",
+      "default": 0
+    },
+    {
+      "name": "age",
+      "type": ["null", "int"]
+    },
+    {
+      "name": "sequence_id",
+      "type": ["null", "long"]
+    },
+    {
+      "name": "name",
+      "type": ["null", "string"],
+      "default": null
+    },
+    {
+      "name": "score",
+      "type": ["null", "float"]
+    },
+    {
+      "name": "avg_score",
+      "type": ["null", "double"]
```

Comment:
Should we also add a nullable type where the default is not null, e.g. `["double", "null"]`? My assumption would be that this defaults to `0.0` instead of `null`. 

According to the [Avro docs on unions](https://avro.apache.org/docs/1.10.2/spec.html#Unions) this is uncommon but allowed.

Reply:
Good call! Added a testcase